packages/@aws-cdk/aws-iam/lib/private/imported-role.ts

```diff
@@ -43,7 +48,7 @@ export class ImportedRole extends Resource implements IRole, IComparablePrincipa
 
   public addToPrincipalPolicy(statement: PolicyStatement): AddToPrincipalPolicyResult {
     if (!this.defaultPolicy) {
-      this.defaultPolicy = new Policy(this, this.defaultPolicyName ?? 'Policy');
+      this.defaultPolicy = new Policy(this, this.defaultPolicyName);
```

Comment:
This now creates very unreadable policy names like `ImportedRolePolicyRoleStack1ImportedRole265D9D4DFE655FCF` (note the duplication of `ImportedRole1`).

I reckon this would work as well. What do you think?

```suggestion
      this.defaultPolicy = new Policy(this, this.defaultPolicyName, {
        policyName: this.defaultPolicyName
      });
```

Reply:
Good idea, will do!

Reply:
After looking even more into this, I'm unsure. Just setting `policyName` to `this.defaultPolicyName` would not make a difference, as the `policyName` anyways defaults to the scope (which is ugly because it is set based on `Names.uniqueId(this)`). If I just set it to a hardcoded value like `'DefaultPolicy'` then this would break, as the name needs to be unique and the problem this PR is supposed to fix is that when you import a Role in multiple stacks, the policy is non-unique, so it would 

The deeper problem is to find a unique readable name. Currently, I'm using `` `Policy${Names.uniqueId(this)}` ``, but maybe there is a better way to create a more readable but still unique name?

Reply:
Pretty sure the default implementation uses the full logical Id of the node as policy name, which is where the duplication originates from. I'm open for other suggestion, but if you run the code you should see what I mean.

PS: Just applying my suggestion wouldn't work though as the code as is would also change how passing an explicit `defaultPolicyName` works. Maybe the flag check should go into `generatePolicyName` 🤔

Reply:
Not sure I can follow. Let's say I have the following code, which defaultPolicyNames would you expect for each role?

```
const app = new App()
const stack1 = new Stack(app, "Stack1")
const roleInStack1 = Role.fromRoleName(stack1, "Role", "RoleName")
const stack2 = new Stack(app, "Stack2")
const roleInStack2 = Role.fromRoleName(stack2, "Role", "RoleName")
```

Reply:
For the above code, I'd like to see something like: `PolicyStack1RoleXXXXXXXX` & `PolicyStack2RoleXXXXXXXX`

---

With the current implementation, we would get:
 `RolePolicyStack1RoleXXXXXXXX` & `RolePolicyStack2RoleXXXXXXXX`
which doesn't seem too bad until you change the Id from `Role` to `LimitWriteAccessForFactoryThingyInterface` because then the name would be `LimitWriteAccessForFactoryThingyInterfacePolicyStack1LimitWriteAccessForFactoryThingyInterfaceXXXXXXX`

Reply:
Thanks! Do you know how I could get `RolePolicyStack1RoleXXXXXXXX`? This doesn't sound like an IAM specific problem, I would assume it is in a utility module like `Names`, but couldn't find anything better than what I'm using so far. If it doesn't exist and is worth it, I could also try to add it to `Names` (or whatever is the right utility module).

Reply:
The string `PolicyStack1RoleXXXXXXXX` you can get by doing `Policy${Names.uniqueId(this)}`, which is exactly how you are doing it at the moment.

Making sure that this string ends up being the policy name is different. I've posted a suggestion at the start of this thread, but it has some implications that need to be worked around. I'm sure there are other ways to achieve this.

Reply:
Ok, I think I got it, and I pushed two commits which should achieve this.